### PR TITLE
Add reader for .exr images

### PR DESCRIPTION
**Pull request by Meakk**
Add a reader for EXR files using OpenEXR library.  
It's enabled using `F3D_MODULE_EXR` option and the reader is automatically registered using the image reader factory.

### CODE REVIEW

File: library/VTKExtensions/Readers/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,16 @@ set(classes
   vtkF3DPostProcessFilter
   )
 
+if(F3D_MODULE_EXR)
+  find_package(OpenEXR REQUIRED)
```
Comment (mwestphal) [inline library/VTKExtensions/Readers/CMakeLists.txt:7]: version check ?

File: library/VTKExtensions/Readers/vtkF3DEXRReader.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  int GetWidth() const;
+  int GetHeight() const;
+
+private:
+  vtkF3DEXRReader(const vtkF3DEXRReader&) = delete;
+  void operator=(const vtkF3DEXRReader&) = delete;
+};
+
+#endif
```
Comment (mwestphal) [inline library/VTKExtensions/Readers/vtkF3DEXRReader.h:35]: missing eol

File: library/VTKExtensions/Readers/vtkF3DEXRReader.cxx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    return;
+  }
+}
+
+int vtkF3DEXRReader::GetWidth() const
+{
+  return this->DataExtent[1] - this->DataExtent[0] + 1;
+}
+
+int vtkF3DEXRReader::GetHeight() const
```
Comment (mwestphal) [inline library/VTKExtensions/Readers/vtkF3DEXRReader.cxx:143]: //------------------------------------------------------------------------------

File: library/VTKExtensions/Readers/vtkF3DEXRReader.cxx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    Imath::Box2i dw = file.dataWindow();
+    this->DataExtent[0] = dw.min.x;
+    this->DataExtent[1] = dw.max.x;
+    this->DataExtent[2] = dw.min.y;
+    this->DataExtent[3] = dw.max.y;
+
+    Imf::RgbaChannels channels = file.channels();
+    if (channels != Imf::RgbaChannels::WRITE_RGBA && channels != Imf::RgbaChannels::WRITE_RGB)
+    {
+      throw std::runtime_error("type of channels not supported yet");
```
Comment (mwestphal) [inline library/VTKExtensions/Readers/vtkF3DEXRReader.cxx:52]: what type of channel do we not support ?

File: library/VTKExtensions/Readers/CMakeLists.txt (hunk 1 above)
Comment (Meakk) [reply library/VTKExtensions/Readers/CMakeLists.txt:7]: why?

File: library/VTKExtensions/Readers/vtkF3DEXRReader.cxx (hunk 4 above)
Comment (Meakk) [reply library/VTKExtensions/Readers/vtkF3DEXRReader.cxx:52]: There's a bunch of different formats in OpenEXR but I've only seen RGB or RGBA files for now.

File: library/VTKExtensions/Readers/CMakeLists.txt (hunk 1 above)
Comment (mwestphal) [reply library/VTKExtensions/Readers/CMakeLists.txt:7]: are we compatible with all existing version of OpenEXR ?

File: testing/data/DATA_LICENSES.md [hunk 5]
```diff
@@ -20,6 +20,7 @@
 - iflamigm.3ds: VTK Data: BSD-3-Clause
 - IM-0001-1983.dcm: VTK Data: BSD-3-Clause
 - InterpolationTest.glb: glTF-Sample-Models: Public Domain
+- kloofendal_43d_clear_1k.exr: Polyhaven: Public Domain
```
Comment (mwestphal) [inline testing/data/DATA_LICENSES.md:23]: arent polyhaven licenses all CC0 ?

File: library/VTKExtensions/Readers/CMakeLists.txt (hunk 1 above)
Comment (Meakk) [reply library/VTKExtensions/Readers/CMakeLists.txt:7]: It's hard to tell.  
Ubuntu 20.04 has version 2.3  
Should I try this one and set it as the minimum version if it works?

File: testing/data/DATA_LICENSES.md (hunk 5 above)
Comment (Meakk) [reply testing/data/DATA_LICENSES.md:23]: The HDR file we are using in the tests is coming from polyhaven too. Should I change its license to cc0?

Comment (Meakk) [reply testing/data/DATA_LICENSES.md:23]: > Our assets are all licensed as [CC0](https://creativecommons.org/publicdomain/zero/1.0/), which is effectively [Public Domain](https://en.wikipedia.org/wiki/Public_domain) even in jurisdictions that do not support the Public Domain.

Comment (mwestphal) [reply testing/data/DATA_LICENSES.md:23]: Yeah, lets update them all to CC0, no reason not to.

File: library/VTKExtensions/Readers/CMakeLists.txt (hunk 1 above)
Comment (mwestphal) [reply library/VTKExtensions/Readers/CMakeLists.txt:7]: Debian 10 has 2.2.1, lets try that ?

File: library/VTKExtensions/Readers/Testing/TestF3DEXRReader.cxx [hunk 6]
```diff
@@ -0,0 +1,31 @@
+#include <vtkNew.h>
+
```
Comment (mwestphal) [inline library/VTKExtensions/Readers/Testing/TestF3DEXRReader.cxx:2]: remove space ?

File: library/VTKExtensions/Readers/Testing/TestF3DEXRReader.cxx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  reader->Print(cout);
+
+  vtkImageData* img = reader->GetOutput();
+
+  int* dims = img->GetDimensions();
+
+  if (dims[0] != 1024 && dims[1] != 512)
+  {
+    std::cerr << "Incorrect EXR image size." << std::endl;
+
```
Comment (mwestphal) [inline library/VTKExtensions/Readers/Testing/TestF3DEXRReader.cxx:26]: remove space(s) ?

**Review by mwestphal — APPROVED** [review]

**Pull request merged**